### DISCUSSION TITLE
Cropping dataset with rotated poles

### DISCUSSION THREAD

**Discussion by ClimateYukon** [pydata/xarray#8082] [Q&A] [answered]
Hello,

I have been looking for a while without much success, I am wondering how to transform a xarray dataset showing a rotated pole into normal lat lon. I am trying to compare a few climate models and some of them such CORDEX or RDRS are using rotated poles. 

One example of dataset would be : 
![chrome_3KM9XZXYNn](https://github.com/pydata/xarray/assets/12923598/66f3f3f1-c30f-4774-95e1-24a1b38b7be2)

Is there an easy way to do this? My goal is to crop it with a shapefile later down the process but I figured the "derotation" should be the first step?

Any help appreciated, 

Thank you so much

**Comment by kmuehlbauer** [comment, marked as answer]
Re-gridding to 1D `lat`/`lon` always involves interpolation and that is sometimes not really wanted. There is https://github.com/pydata/xarray/issues/2281 discussing potential approaches. 

You might also use the provided 2D `lat`/`lon` coordinates for cropping/masking. As always this depends on your actual use-case. The below example just crops with a simple bounding box.

```python
ds_clip = ds.where(
    (
        ((ds.lat > bbox["bottom"]) & (ds.lat < bbox["top"]))
        & ((ds.lon > bbox["left"]) & (ds.lon < bbox["right"]))
    ),
    drop=True,
)
```

Another approach would be to project your shapefile coordinates into the rotated pole domain and do cropping and all further computation in that domain. This at least excludes any errors which might be introduced by grid interpolation.

**Reply by ClimateYukon** [reply]
Thank you! I will give it a try.
